app/graph/types/task_type.rb

```diff
@@ -1,58 +1,57 @@
-TaskType = GraphqlCrudOperations.define_annotation_type('task', { label: 'str', type: 'str', annotated_type: 'str', description: 'str', json_schema: 'str', slug: 'str' }) do
-  field :first_response do
-    type AnnotationType
-
-    resolve -> (task, _args, _ctx) {
-      obj = task.load || task
-      obj.nil? ? nil : obj.first_response_obj
-    }
+class TaskType < AnnotationObject
+  define_shared_behavior(self, 'task')
+
+  field :label, String, null: true
+  field :type, String, null: true
+  field :annotated_type, String, null: true
+  field :description, String, null: true
+  field :json_schema, String, null: true
+  field :slug, String, null: true
+
+  field :first_response, "AnnotationType", null: true
```

Comment:
TODO: DO we need this in a string?

Reply:
we seem to; get a circular reference for annotation type if we don't. still not sure why